zerver/tests/test_bots.py

```diff
@@ -1179,6 +1179,162 @@ def test_patch_bot_full_name_non_bot(self) -> None:
+
+        # Create second bot
+        bot_info = {
+            "full_name": "Bot 2",
+            "short_name": "bot2",
+        }
+        result = self.client_post("/json/bots", bot_info)
+        self.assert_json_success(result)
+        bot2 = self.get_bot_user("bot2-bot@zulip.testserver")
+
+        # Try to change bot2's short_name to bot1's
+        bot_info = {
+            "short_name": "bot1",
+        }
+        result = self.client_patch(f"/json/bots/{bot2.id}", bot_info)
+        self.assert_json_error(result, "Email address already in use")
+
+    def test_patch_bot_short_name_invalid(self) -> None:
+        self.login("hamlet")
+        self.create_bot()
+        email = "hambot-bot@zulip.testserver"
+        bot = self.get_bot_user(email)
+
+        # Empty short_name should fail
+        bot_info = {
+            "short_name": "",
+        }
+        result = self.client_patch(f"/json/bots/{bot.id}", bot_info)
+        self.assert_json_error(result, "Bad name or username")
+
+    def test_patch_bot_short_name_no_change(self) -> None:
+        self.login("hamlet")
+        self.create_bot()
+        email = "hambot-bot@zulip.testserver"
+        bot = self.get_bot_user(email)
+
+        # Changing to the same short_name should be a no-op
+        bot_info = {
+            "short_name": "hambot",
+        }
+        result = self.client_patch(f"/json/bots/{bot.id}", bot_info)
+        self.assert_json_success(result)
+
+        bot.refresh_from_db()
+        self.assertEqual(bot.email, "hambot-bot@zulip.testserver")
+
+    def test_patch_bot_short_name_invalid_fake_email_domain(self) -> None:
+        self.login("hamlet")
+        # Create bot first without the invalid domain setting
+        self.create_bot()
+        email = "hambot-bot@zulip.testserver"
+        bot = self.get_bot_user(email)
+
+        # Now try to change short_name with invalid FAKE_EMAIL_DOMAIN
+        with override_settings(
+            FAKE_EMAIL_DOMAIN="invaliddomain", REALM_HOSTS={"zulip": "127.0.0.1"}
+        ):
+            bot_info = {
+                "short_name": "newbot",
+            }
+            result = self.client_patch(f"/json/bots/{bot.id}", bot_info)
+            error_message = (
+                "Can't change bot email until FAKE_EMAIL_DOMAIN is correctly configured.\n"
+                "Please contact your server administrator."
+            )
+            self.assert_json_error(result, error_message)
+
+    def test_patch_bot_short_name_unexpected_email_format(self) -> None:
+        """Test the fallback case when bot email doesn't end with '-bot'."""
+        self.login("hamlet")
+        self.create_bot()
+        email = "hambot-bot@zulip.testserver"
+        bot = self.get_bot_user(email)
+
+        # Manually set bot email to an unexpected format (without -bot suffix)
+        # This tests the fallback case in the code
+        bot.email = "unexpected@zulip.testserver"
```

Comment:
I don't think we should test this. We do not test such cases where a field is set to an invalid value by mnaully updating in DB.

Reply:
You haven't removed this.